tests/Feature/Platform/Action/UpdatePlayerGameMetricsTest.php

```diff
@@ -0,0 +1,322 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature\Platform\Action;
+
+use App\Models\Achievement;
+use App\Models\PlayerAchievement;
+use App\Models\PlayerGame;
+use App\Models\User;
+use App\Platform\Actions\UpdateGameMetrics;
+use App\Platform\Actions\UpdatePlayerGameMetrics;
+use App\Platform\Enums\AchievementType;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\Feature\Platform\Concerns\TestsPlayerAchievements;
+use Tests\Feature\Platform\Concerns\TestsPlayerBadges;
+use Tests\TestCase;
+
+class UpdatePlayerGameMetricsTest extends TestCase
+{
+    use RefreshDatabase;
+
+    use TestsPlayerAchievements;
+    use TestsPlayerBadges;
+
+    public function testMetrics(): void
+    {
+        User::factory()->count(1)->create();
+
+        $user = User::first();
```

Comment:
```suggestion
        $user = User::factory->create();
```
nit: Less ambiguous to combine these. Some tests are seeded with an initial user (`$this->user`) which could make the intent here for a future maintainer unclear.